Fix build failures on macOS

See https://github.com/swesonga/Info-ZIP/runs/4612311935?check_suite_focus=true

```
cc -c -I. -DUNIX -O3 -DBZIP2_SUPPORT -DUIDGID_NOT_16BIT -DLARGE_FILE_SUPPORT -DUNICODE_SUPPORT -DNO_RMDIR -DNO_STRCHR -DNO_STRRCHR -DNO_RENAME -DNO_MKTEMP -DNO_MKTIME -DNO_MKSTEMP -DZMEM -DNO_DIR -DHAVE_DIRENT_H -DHAVE_TERMIOS_H zip.c
In file included from zip.c:16:
./zip.h:716:10: error: conflicting types for 'memset'
   char *memset OF((char *, int, unsigned int));
         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/string.h:74:7: note: previous declaration is here
void    *memset(void *__b, int __c, size_t __len);
         ^
In file included from zip.c:16:
./zip.h:717:10: error: conflicting types for 'memcpy'
   char *memcpy OF((char *, char *, unsigned int));
         ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/string.h:72:7: note: previous declaration is here
void    *memcpy(void *__dst, const void *__src, size_t __n);
         ^
In file included from zip.c:16:
./zip.h:718:8: error: conflicting types for 'memcmp'
   int memcmp OF((char *, char *, unsigned int));
       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/string.h:71:6: note: previous declaration is here
int      memcmp(const void *__s1, const void *__s2, size_t __n);
         ^
```

--- a/zip30/unix/configure
+++ b/zip30/unix/configure
@@ -519,7 +519,10 @@ done
 
 
 echo Check for memset
-echo "int main(){ char k; memset(&k,0,0); return 0; }" > conftest.c
+cat > conftest.c << _EOF_
+#include <string.h>
+int main(){ char k; memset(&k,0,0); return 0; }
+_EOF_
 $CC -o conftest conftest.c >/dev/null 2>/dev/null
 [ $? -ne 0 ] && CFLAGS="${CFLAGS} -DZMEM"
 

--- a/zip30/unix/unix.c
+++ b/zip30/unix/unix.c
@@ -67,7 +67,6 @@ local char *readd OF((DIR *));
 #ifndef dirent
 #  define dirent direct
 #endif
-typedef FILE DIR;
 /*
 **  Apparently originally by Rich Salz.
 **  Cleaned up and modified by James W. Birdsall.

